slides/week0/week0.org

```diff
@@ -0,0 +1,484 @@
+  }
+#+END_SRC
+
+* Click Run
+#+ATTR_HTML: :width 50%
+[[file:https://i.imgur.com/caYg2AY.png]]
+
+* {}
+- Curly Braces define scope of a function in the code
+- Right now that means that your code goes in the main function
+- All compound or control flow statements have curly braces
+    - Below the *{}* defines the main method
+#+BEGIN_SRC C++ -n
+  int main() {
+    // <==== HERE =====
+  }
+#+END_SRC
+
+* ;
+#+BEGIN_NOTES
+Make sure to note that they will see this error a lot and that it will go away with time.
+#+END_NOTES
+- A semicolon defines the end of a line of code
+- Think of it like a period in a sentence
+- All expression statements end in a semicolon
+    - declarations, assignments, function calls
+    - A missing semicolon results in an error message that can point to the wrong line
+#+BEGIN_SRC C++ -n
+    #include <iostream>
+    using namespace std;
+    int main() {
+      cout << "Hello" // <=====
+      cout << "World!";
+      return 0;
+    }
+#+END_SRC
+- Output
+#+BEGIN_EXAMPLE
+/temp/file.cpp: In function ‘int main()’:
+/temp/file.cpp:5:5: error: expected ‘;’ before ‘cout’
+     cout << "World!";
+     ^
+
+Compilation Failed
+#+END_EXAMPLE
+
+* Looking back at Hello World!
+- our code is in {}
+- *cout* tells the computer to write the letters to the screen
+- *<<* tells the computer what to write out
+- line 1: inclues a header file (covered later)
+- line 4: defines the main function
+- line 6: tells the computer that the program ran successfully
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  #include <iostream>
+  using namespace std;
+
+  int main() {
+    cout << "Hello World!";
+    return 0;
+  }
+#+END_SRC
+
+* Most useful program ever
+#+BEGIN_NOTES
+This should naturally lead into why variables are useful. Without them
+all programs would be deterministic.
+#+END_NOTES
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  int main() {
+    cout << 2 + 2;
+    return 0;
+  }
+#+END_SRC
+
+* Variables
+- Alias for a value
+- the value can be changed without changing the alias
+
+* Primitive Variables Types
+| Name | Description |
+|-------+------|
+| =bool= | Can be either *true* or *false*|
+| =char= | a character like 'c' or '+'|
+| =int= | A whole number *bounded|
+| =double= | a decimal number *bounded|
+- What makes a variable primitive?
+
+* Primitive Variables?
+[[file:https://i.imgur.com/Wo0ovX5.jpg]]
+- But Actually... It means that it cannot be divided into smaller parts
+- A number is just a number
+
+* Important Symbols
+#+BEGIN_NOTES
+No lengthy explanation of << and stdout
+#+END_NOTES
+| symbol | meaning |
+|-------+------|
+| =//= | Signifies a comment, The computer will not try to compile that line |
+| === | assignment operator, value of right is assigned to variable on left |
+| =<<= | sends something to the OS which displays it in the terminal |
+
+* Math Operators
+- order of operations still applies
+- ~()~ > ~*,/~ > ~+,-~
```

Comment:
This line just looks like a bunch of random characters (even rendered.) I know it's supposed to show order of operations but it's not at all clear.

Maybe just remove this and order the table by order of operations?